src/core/Akka/Actor/ActorCell.FaultHandling.cs

```diff
@@ -67,9 +67,10 @@ private void FaultRecreate(Exception cause)
                     
                     failedActor.AroundPreRestart(cause, optionalMessage);
 
-                    // run actor pre-incarnation plugin pipeline
-                    var pipeline = _systemImpl.ActorPipelineResolver.ResolvePipeline(failedActor.GetType());
-                    pipeline.BeforeActorIncarnated(failedActor, this);
+                    if (failedActor is IActorStash stashed)
+                    {
+                        stashed.Stash.UnstashAll();
```

Comment:
Is the idea here to include automatic stash / unstashing upon actor restart?

Reply:
This is not my idea - I'm just keeping the original behavior ;)

Reply:
Not in the Akka.NET codebase - open a new PR for this.

Reply:
See: https://github.com/akkadotnet/akka.net/pull/3862/files#diff-a2c304463bb7ef467388df75c265bbfbL63

Reply:
I stand corrected! You're right.

https://github.com/akkadotnet/akka.net/blob/cf0cb8c1daffc68b8673a491759416cae47f6af1/src/core/Akka.Tests/Actor/Stash/ActorWithStashSpec.cs#L90-L111